### PR TITLE
feat(pkg) add 'mirrorbrain' user back under puppet management

### LINKED ISSUE #3351: [pkg.origin.jenkins.io] User `mirrorbrain` is unmanaged
Since https://github.com/jenkins-infra/jenkins-infra/pull/2185 the legacy service `mirrors.jenkins.io` has been migrated to [`get.jenkins.io`](https://github.com/jenkins-infra/kubernetes-management/blob/06a8d59087b50bdde93e15c64968efdd2b5113ec/clusters/publick8s.yaml#L196-L203).

But I made the change #2185 too hastily and since then, the user `mirrorbrain` ([used by the update center and core releases processes to deploy content to `pkg.jenkins.io` / `updates.jenkins.io` VM](https://github.com/jenkins-infra/helpdesk/issues/3414#issuecomment-2019574326)) is unmanaged.

This user is actively used so it MUST be managed, including its SSH keys (public/private) and the crontab running the `sync.sh` script (at least!), not mentioning the `/srv/release` management.

Note: This user should be renamed in the future to make it distinct from the legacy setup but it might be wise to wait for https://github.com/jenkins-infra/helpdesk/issues/3705 and https://github.com/jenkins-infra/helpdesk/issues/2649

Ref. https://github.com/jenkins-infra/helpdesk/issues/2970

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/3414#issuecomment-2025472147

This PR fixes #3351 by putting back the `mirrorbrain` user under puppet management and also manages the `.azure-storage-env` with the current values from resources managed in https://github.com/jenkins-infra/azure/blob/main/get.jenkins.io.tf.

Added resources comes from https://github.com/jenkins-infra/jenkins-infra/pull/2185.

Tested locally with vagrant but of course there could be a configuration shift with production: before merging, the puppet agent on `pkg` VM must be disabled and stopped. Once merged, a `puppet` noop command is needed to compare the expected changes (and avoid breaking the current setup which is manually managed).

Note that a manual cleanup is required after this PR is applied. The following files will have to be removed:

- `/srv/releases/.ssh/gruenehoelle_mirror*`  (ref. https://github.com/jenkins-infra/helpdesk/issues/3281)
- `/srv/releases/.ssh/mirrorbits-staging*` (not used anywhere)

### CODE REVIEW

**Comment by dduportal** [comment]
Puppet agent disabled on the `pkg` VM. But please don't merge if approved.

File: dist/profile/manifests/pkgrepo.pp [hunk 1]
```diff
@@ -66,7 +69,73 @@
...
+export AZURE_STORAGE_KEY=${lookup('azure::getjenkinsio::storagekey')}
+",
+    require => [
+      Account[$mirror_user],
+    ],
+  }
+
+  cron { 'mirrorbrain-sync-releases':
+    command => "cd ${mirror_home_dir} && time ./sync.sh --full-sync > last_sync.log",
+    # 0 * * * * cd /srv/releases && time ./sync.sh --full-sync > last_sync.log
```
Comment (lemeurherve) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
```

File: dist/profile/manifests/pkgrepo.pp [hunk 2]
```diff
@@ -95,6 +164,7 @@
       group   => $mirror_group,
     }
   }
+  ############################################################################################################
```
Comment (lemeurherve) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
  ################################################################################################
```
(nit, same size as below)

File: dist/profile/manifests/pkgrepo.pp [hunk 3]
```diff
@@ -51,6 +51,9 @@
     ensure => 'installed',
   }
 
+  ############################################################################################################
+  # Mirrorbrain User management
+  ############################################################################################################
```
Comment (lemeurherve) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
  ################################################################################################
  # Mirrorbrain User management
  ################################################################################################
```
(nit, same size as below)

**Review by lemeurherve — DISMISSED** [review]
🎉

**Review by lemeurherve — DISMISSED** [review]

**Comment by dduportal** [comment]
Deployment result:

- ✅ Puppet agent execution reported:

```
Mar 29 06:39:00 ip-172-31-33-96 puppet-agent[13349]: (/Stage[main]/Profile::Pkgrepo/File[/var/www/pkg.jenkins.io/redhat-stable/jenkins.io.key]/content) content changed <redacted> (corrective)
Mar 29 06:39:01 ip-172-31-33-96 puppet-agent[13349]: (/Stage[main]/Profile::Pkgrepo/File[/srv/releases/.ssh/archives]/content) content changed <redacted>
Mar 29 06:39:01 ip-172-31-33-96 puppet-agent[13349]: (/Stage[main]/Profile::Pkgrepo/File[/srv/releases/.ssh/config]/content) content changed <redacted>
Mar 29 06:39:01 ip-172-31-33-96 puppet-agent[13349]: (/Stage[main]/Profile::Pkgrepo/File[/srv/releases/.azure-storage-env]/group) group changed 'root' to 'mirrorbrain'
Mar 29 06:39:01 ip-172-31-33-96 puppet-agent[13349]: (/Stage[main]/Profile::Pkgrepo/File[/srv/releases/.azure-storage-env]/mode) mode changed '0644' to '0600'
```

- ✅ Tried changing the `mirrorbrain` crontab: the 2nd puppet agent execution removed my change (as expected)

- ✅ `sync.sh` manual execution:

```
+ echo '>> Script sync.sh finished'
+ exit 0

real    1m38.052s
user    1m44.130s
sys     0m19.152s
```

- ✅ Update center ran without any issues

**Pull request merged**